adding file source page (#11355)
opensearch-project/documentation-website: _data-prepper/pipelines/configuration/sources/file.md

---
layout: default
title: File
parent: Sources
grand_parent: Pipelines
nav_order: 24
---

# File source

> The `file` plugin reads events from a local file once at pipeline startup. It is ideal for loading seed data, testing processors and sinks, or replaying a fixed dataset. This source **does not tail** a file for new lines after startup.

Reviewer: ```suggestion
The `file` plugin reads events from a local file once when the pipeline starts. It's useful for loading seed data, testing processors and sinks, or replaying a fixed dataset. This source *does not monitor* the file for new lines after startup.
```

Option | Required | Type | Description
:--- | :--- | :--- | :---

> `path` | Yes | String | Absolute path to the input file inside the Data Prepper container, for example, `/usr/share/data-prepper/data/input.jsonl`.

Reviewer: ```suggestion
`path` | Yes | String | An absolute path to the input file inside the Data Prepper container, for example, `/usr/share/data-prepper/data/input.jsonl`.
```
